cloudvolume/datasource/precomputed/mesh/sharded.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+
+from ..sharding import ShardingSpecification, ShardReader
+from ....storage import SimpleStorage
+from ....mesh import Mesh
+from ....lib import red
+
+class ShardedMultiLevelPrecomputedMeshSource:
+    def __init__(self, meta, cache, config, readonly=False):
+        self.meta = meta
+        self.cache = cache
+        self.config = config
+        self.readonly = bool(readonly)
+
+        spec = ShardingSpecification.from_dict(self.meta.info['sharding'])
+        self.reader = ShardReader(meta, cache, spec)
+
+        self.vertex_quantization_bits = self.meta.info['vertex_quantization_bits']
+        self.lod_scale_multiplier = self.meta.info['lod_scale_multiplier']
+        self.transform = np.array(self.meta.info['transform'] + [0,0,0,1]).reshape(4,4)
+    
+        if np.any(self.transform * np.array([[0,1,1,1],[1,0,1,1],[1,1,0,1],[1,1,1,0]])):
+            raise ValueError(red("Non-scale homogeneous transforms are not implemented"))
+
+    @property
+    def path(self):
+        return self.meta.mesh_path
+    
+    def get(self, segids, lods=None):
+        """Fetch meshes at all levels of details.
+
+        Parameters:
+        segids : int or [int ..]
+            Scalar or list of integer segment IDs.
+        lods : int, [int, ..] or None
+            Level of detail(s) to retrieve.  0 is highest level of detail.
+            None will fetch meshes at all levels.
+
+        Returns:
+        [Mesh, ..] or [[Mesh, ..], ..]
+            A list of meshes for each ssegment ID is returned.
+            An additional layer is added when multiple lods are requested.
+
+        Reference:
+            https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md
+        """
+        seg_list_return = True
+        if type(segids) in (int, float):
+            seg_list_return = False
+            segids = [ int(segids) ]
+
+        results = []
+        for segid in segids:
+            # Read the manifest (with a tweak to sharding.py to get the offset)
+            binary, shard_file_offset = self.reader.get_data(segid, self.meta.mesh_path, return_offset=True)
+            manifest = MultiLevelPrecomputedMeshManifest(binary)
+
+            single_lod_return = False
+            if isinstance(lods, int):
```

Comment:
For this, you can also use `toiter(lods, is_iter=True)` to get both an iterable lods and whether the input was a scalar.